src/devices/video/upd7220.cpp

```diff
@@ -1541,6 +1579,46 @@ uint8_t upd7220_device::dack_r()
 
 void upd7220_device::dack_w(uint8_t data)
 {
+	switch(m_dma_type) {
+	case 0:
+		if (m_dma_transfer_length % 2) {
+			m_dma_data = ((m_dma_data & 0xff) | data << 8) & m_mask;
+			write_vram(m_dma_type, m_dma_mod, m_dma_data);
+		} else {
+			m_dma_data = (m_dma_data & 0xff00) | data;
+		}
+		break;
+	case 2:
+		m_dma_data = data & (m_mask & 0xff);
+		write_vram(m_dma_type, m_dma_mod, m_dma_data);
+		break;
+	case 3:
+		m_dma_data = (data << 8) & (m_mask & 0xff);
```

Comment:
This can’t possibly be right – it always evaluates to zero (you’re shifting left by eight bits, then computing a union with a mask restricted to the low eight bits).